test/unit/math/rev/mat/vectorize/apply_scalar_unary_test.cpp

```diff
@@ -0,0 +1,18 @@
+#include <stan/math/rev/core/var.hpp>
+#include <stan/math/rev/scal/fun/exp.hpp>
+#include <stan/math/rev/mat/vectorize/apply_scalar_unary.hpp>
+#include <test/unit/math/prim/mat/vectorize/foo_base_test.hpp>
+#include <test/unit/math/prim/mat/vectorize/expect_types.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_values.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_errors.hpp>
+#include <gtest/gtest.h>
+
+/**
+ * this tests that the expect_values test works on a mock function 
+ */
+TEST(MathRevMatVectorize, applyScalarUnaryMock) {
```

Comment:
You need to add header guards to anything you include.  That'll
prevent the same code from getting included twice.

But if you're redefining the same function signature, it's
probably because you were using too-general types.
Rather than (for some return type R):

  template <typename T>
  void foo(const T& x) {
    ... code to treat x as matrix ..

use

  template <typename T, int R, int C>
  void foo(const Eigen::Matrix<T, R, C>& x) {
    ... same code as before to treat x as matrix ...

- Bob

> On Mar 4, 2016, at 4:17 PM, rayleigh <notifications@github.com> wrote:
> 
> In test/unit/math/rev/mat/vectorize/apply_scalar_unary_test.cpp:
> 
> > @@ -0,0 +1,18 @@
> > +#include <stan/math/rev/core/var.hpp>
> > +#include <stan/math/rev/scal/fun/exp.hpp>
> > +#include <stan/math/rev/mat/vectorize/apply_scalar_unary.hpp>
> > +#include <test/unit/math/prim/mat/vectorize/foo_base_test.hpp>
> > +#include <test/unit/math/prim/mat/vectorize/expect_types.hpp>
> > +#include <test/unit/math/rev/mat/vectorize/expect_values.hpp>
> > +#include <test/unit/math/rev/mat/vectorize/expect_errors.hpp>
> > +#include <gtest/gtest.h>
> > +
> > +/**
> > + * this tests that the expect_values test works on a mock function 
> > + */
> > +TEST(MathRevMatVectorize, applyScalarUnaryMock) {
> 
> Thanks for showing me how to do it yesterday. I'm trying what you suggested, but I'm getting errors because the underlying test functions are named the same and are being redefined due to the include statements. I was worried that I would run into this issue if we wanted a mix function calling all tests.
> 
> Today, I tried adding a stan::math::var as a dummy template parameter, but that caused other issues. I could:
> 
> a) Add a dummy argument to the function call (e.g. expect_scalar_value(stan::math::var x)) 
> b) Add a different number of dummy template parameters using int
> c) Rename functions
> d) Some other approach
> 
> I'm wondering what the best would be.
> 
> —
> Reply to this email directly or view it on GitHub.
>